docs/recipes/ReadUintFromContract.md

```diff
@@ -0,0 +1,197 @@
+        <h2 className="card-title">Greetings Count</h2>
+        <div className="card-actions items-center flex-col gap-1 text-lg">
+          <h2 className="font-bold m-0">Total Greetings count:</h2>
+          {isTotalCounterLoading ? (
+            <span className="loading loading-spinner"></span>
+          ) : (
+            <p className="m-0">{totalCounter ? totalCounter.toString() : 0}</p>
+          )}
+          <h2 className="font-bold m-0">Your Greetings count:</h2>
+          {isConnectedAddressCounterLoading ? (
+            <span className="loading loading-spinner"></span>
+          ) : (
+            <p className="m-0">{connectedAddressCounter ? connectedAddressCounter.toString() : 0}</p>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+};
+```
+
+</details>
+
+## Implementation guide
+
+### Step 1: Create a new Component
+
+Begin by creating a new component in the "components" folder of your application.
+
+```tsx title="components/GreetingsCount.tsx"
+export const GreetingsCount = () => {
+  return (
+    <div>
+      <h2 className="font-bold m-0">Total Greetings count:</h2>
+      <h2 className="font-bold m-0">Your Greetings count:</h2>
+    </div>
+  );
+};
+```
+
+### Step 2: Retrieve total greetings count
+
+Initialize the [useScaffoldContractRead](/hooks/scaffold-eth#usescaffoldcontractread) hook to read from contract. This hook provides the `data` which contains the return value of the function.
+
+```tsx title="components/GreetingsCount.tsx"
+//highlight-start
+import { useScaffoldContractRead } from "~~/hooks/scaffold-eth";
+// highlight-end
+
+export const GreetingsCount = () => {
+  // highlight-start
+  const { data: totalCounter } = useScaffoldContractRead({
+    contractName: "YourContract",
+    functionName: "totalCounter",
+    watch: true,
+  });
+  // highlight-end
+
+  return (
+    <div>
+      <h2 className="font-bold m-0">Total Greetings count:</h2>
+      //highlight-start
+      <p>{totalCounter ? totalCounter.toString() : 0}</p>
+      //highlight-end
+      <h2 className="font-bold m-0">Your Greetings count:</h2>
+    </div>
+  );
+};
+```
+
+`const {data: totalCounter} = useScaffoldContractRead({...})` here we are using [destructuring asssignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to assign `data` to new name `totalCounter`.
+
+`totalCounter` in contract returns an `uint` value, which is represented as [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) in javascript and can be converted to readable string using `.toString()`.
+
+### Step 3: Retrieve connected address greetings count
+
+We can access connected address using [useAccount](https://wagmi.sh/react/hooks/useAccount) hook and pass it to `args` key in `useScaffoldContractRead` hooks configuration which will be used as an argument to the read contract function.
```

Comment:
```suggestion
We can get the connected address using the [useAccount](https://wagmi.sh/react/hooks/useAccount) hook and pass it to `args` key in the `useScaffoldContractRead` hooks configuration. The address will then be used as an argument for reading data from the contract function.
```

^maybe changed the wording too much here, and made it perhaps longer. We can also say "This will be used as an argument to read the contract function." The inital wording felt a bit confusing to me.

Reply:
Yup fixed at [77690fe](https://github.com/scaffold-eth/se2-docs/pull/48/commits/77690fe10d3a712ad431b6c1db6a1fd28ec640fc), thanks 🙌